packages/kolibri-common/composables/useCoachMetadataTags.js

```diff
@@ -1,66 +1,126 @@
-    return contentNode.categories.map(category => coreString(category));
+    return contentNode.categories.map(category => createTag(coreString(category), category));
   };
 
   const getLevelTags = () => {
     if (!contentNode.grade_levels) return [];
-    return contentNode.grade_levels.map(grade_levels => coreString(grade_levels));
+    return contentNode.grade_levels.map(grade_levels =>
+      createTag(coreString(grade_levels), grade_levels),
+    );
   };
 
   const getLanguageTag = () => {
-    if (!contentNode.lang) return [];
-    return contentNode.lang.lang_name;
+    return createTag(contentNode.lang.lang_name, contentNode.lang.id);
   };
 
-  const getActivityTag = () => {
+  const getActivityTags = () => {
     if (!contentNode.learning_activities) return [];
 
-    return contentNode.learning_activities.length > 1
-      ? [coreStrings.$tr('multipleLearningActivities')]
-      : contentNode.learning_activities.map(activity => coreString(activity));
+    if (contentNode.learning_activities.length > 1) {
+      return createTag(
+        coreStrings.$tr('multipleLearningActivities'),
+        'multipleLearningActivities',
+        'allActivities',
+      );
+    } else {
+      return contentNode.learning_activities.map(activity => {
+        let icon;
+        if (activity === LearningActivities.EXPLORE) {
+          icon = 'interactSolid';
+        } else {
+          icon = ActivitiesLookup[activity].toLowerCase() + 'Solid';
+        }
+        return createTag(coreString(activity), activity, icon);
+      });
+    }
   };
 
   const getDurationTag = () => {
     if (!contentNode.duration) return [];
-    return contentNode.duration.map(duration => coreStrings.formatDuration(duration));
+    return [createTag(durationEstimation.value, contentNode.duration)];
   };
 
   const getSpecificCategoryTag = () => {
     if (!contentNode.categories) return [];
     const specificCategories = contentNode.categories.filter(
       category => category.split('.').length > 2,
     );
-    return specificCategories.map(category => coreString(category));
+    return specificCategories.map(category => createTag(coreString(category), category));
+  };
+
+  const getFolderTags = () => {
+    return [getKindTag()];
+  };
+
+  const getResourceTags = () => {
+    return [...getActivityTags(), ...getDurationTag(), ...getCategoryTags()];
+  };
+
+  // Placeholder for possible need to handle lanugage tags gracefully
+  const getChannelTags = () => {
+    return getResourceTags();
   };
 
   if (
     contentNode.kind === ContentNodeKinds.CHANNEL ||
     contentNode.kind === ContentNodeKinds.TOPIC
   ) {
     tags.value = [
+      getKindTag(),
       ...getCategoryTags().slice(0, 3),
       ...getLevelTags().slice(0, 3),
-      ...getLanguageTag(),
+      getLanguageTag(),
     ];
   } else {
     tags.value = [
-      ...getActivityTag(),
-      ...getDurationTag(),
+      ...getActivityTags(),
+      //getDurationTag(),
```

Comment:
Are we not using the duration anywhere?

Reply:
@marcellamaki I've cleaned up unused stuff here.

When I spoke w/ @jtamiace the tags that are returned by the get*Metadata functions are the ones she said we should be returning. I left a couple internal functions in comments as they may be useful in the future if we make design changes.